server/src/main/java/org/opensearch/common/cache/stats/ImmutableCacheStatsHolder.java

```diff
@@ -23,15 +32,92 @@
  */
 
 @ExperimentalApi
-public class ImmutableCacheStatsHolder { // TODO: extends Writeable, ToXContent
-    // An immutable snapshot of a stats within a CacheStatsHolder, containing all the stats maintained by the cache.
+public class ImmutableCacheStatsHolder implements Writeable, ToXContent {
+    // Root node of immutable snapshot of stats within a CacheStatsHolder, containing all the stats maintained by the cache.
     // Pkg-private for testing.
     final Node statsRoot;
+    // The dimension names for each level in this tree.
     final List<String> dimensionNames;
+    // The name of the cache type producing these stats. Returned in API response.
+    final String storeName;
+    public static String STORE_NAME_FIELD = "store_name";
 
-    public ImmutableCacheStatsHolder(Node statsRoot, List<String> dimensionNames) {
-        this.statsRoot = statsRoot;
-        this.dimensionNames = dimensionNames;
+    // Values used for serializing/deserializing the tree.
+    private static final String SERIALIZATION_CHILDREN_OPEN_BRACKET = "<";
+    private static final String SERIALIZATION_CHILDREN_CLOSE_BRACKET = ">";
+    private static final String SERIALIZATION_BEGIN_NODE = "_";
+    private static final String SERIALIZATION_DONE = "end";
+
+    ImmutableCacheStatsHolder(
+        CacheStatsHolder.Node originalStatsRoot,
+        List<String> levels,
+        List<String> originalDimensionNames,
+        String storeName
+    ) {
+        // Aggregate from the original CacheStatsHolder according to the levels passed in.
+        List<String> filteredLevels = filterLevels(levels, originalDimensionNames);
+        this.dimensionNames = filteredLevels; // The dimension names for this immutable snapshot should reflect the levels we aggregate in
+                                              // the snapshot
+        this.storeName = storeName;
+        this.statsRoot = aggregateByLevels(filteredLevels, originalStatsRoot, originalDimensionNames);
```

Comment:
`aggregateByLevels` can directly use the member variable dimensionNames? Also, I will call these variables as `filteredDimensions` and `allDimensions` or anything better you can come up with?